src/jit/morph.cpp

```diff
@@ -12748,24 +12748,22 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                     }
                 }
 
-#ifdef DEBUG
-                // If we have decided to fold, then temp cannot be nullptr
-                if (foldAndReturnTemp)
+                if ((temp != nullptr) && !foldAndReturnTemp)
```

Comment:
This needs a comment. It's a weird condition, and it's not 100% clear why you are only checking this when `!foldAndReturnTemp` so it should be explained (i.e. in that case we know that the load is a valid size).